File: 27-remove-element/remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int size = nums.size();
        int slow = 0; // slow pointer
        
        for (int fast = 0; fast < size; fast++) { // fast pointer
            if (nums[fast] != val) {
                nums[slow] = nums[fast];
                slow++;
            }
        }
        
        return slow;
        
    }
};

File: 27-remove-element/remove-element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int n=nums.length;
        int none=0;
        for(int i=0;i<n;i++)
        {
        
            if(nums[i]!=val)
                nums[none++]=nums[i];
        }
        return none;
        
    }
}